llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -3152,6 +3153,34 @@ class TargetLoweringBase {
     return false;
   }
 
+  /// Lower an interleaved load to target specific intrinsics. Return
+  /// true on success.
+  ///
+  /// \p Load is a vp.load instruction.
+  /// \p Mask is a mask value
+  /// \p DeinterleaveIntrin is vector.deinterleave intrinsic
+  /// \p DeinterleaveRes is a list of deinterleaved results.
+  virtual bool
+  lowerInterleavedScalableLoad(VPIntrinsic *Load, Value *Mask,
```

Comment:
If this doesn't have to be strictly restricted to scalable vectors, would it make sense to rename this something like `lowerDeinterleaveIntrinsicToVPLoad` to more closely match the `lowerDeinterleaveIntrinsicToLoad` hook?

Reply:
I also updated the list of arguments taken by lowerDeinterleaveIntrinsicToVPLoad / lowerInterleaveIntrinsicToVPStore and they no longer take the (de)interleaved intrinsic argument (because we're not really using it)